### ISSUE TITLE
cannot connect to the multipass socket

### ISSUE THREAD

**Issue by GogoFC** [canonical/multipass#2505]
Windows 11
v1.8.1

Error: 

```
PS C:\Users\Goran> multipass list
list failed: cannot connect to the multipass socket
Please ensure multipassd is running and 'localhost:50051' is accessible
```

So this is what  I did:

1. Tried out custom partitioning via user-data, partitions got created but other partitions showed, rebooted VM. VM got stuck couldn't delete or stop it, so I stopped it via Hyper-V manager. Rebooted.
2.  After reboot the error came up.
3.  Reinstalled Multipass, same thing.
4.  Reinstalled Multipass with deleting all of the data, one VM that was still running didn't get deleted, all of the VM's "placeholders' in Hyper-V GUI manager seemed like they exist even though they were deleted. 
5.  Stopped the VM and reinstalled Multipass, VM got deleted but error persisted, can not start multipassd.
6.   Can not reinstall Multipass, about time to Nuke this Installation of Windows soon anyway.
7.   Event Viewer shows nothing, well it shows 6000+ separate events, all fine one repetitve warning.
```
  [url downloader] Cannot retrieve headers for https://cdimage.ubuntu.com/ubuntu-core/16/stable/current/ubuntu-core-16-amd64.img.xz: Host cdimage.ubuntu.com not found  
```
8.  Attempted to start multipassd.exe manually got an error about forbidden D:, I don't have a D drive.  Had to record this, it's not in the Event Viewer. 

![Screenshot 2022-03-31 012113](https://user-images.githubusercontent.com/9360410/160948803-629a6a33-ea1b-438e-9937-b90345b6c117.png)

9.  Deleted leftovers from `%programfiles%` and `%appdata%`, there was some caches left over in Program Files, user appdata local and roaming. Still error persists after reinstall. 
10.  Installer now says "Daemon install result: 0"

Multipass GUI now runs twice, daemon doesn't start, GUI says failure to retrieve instances.

![Screenshot 2022-03-31 015222](https://user-images.githubusercontent.com/9360410/160949320-16580794-f5b2-4856-aa85-d5420c2a1bfa.png)

![Screenshot (133)](https://user-images.githubusercontent.com/9360410/160949349-22c01247-c3f8-4e7d-8038-4babc99329e0.png)

The Screenshot below shows the running VM named 'sad' even after reinstall of Multipass.

![Screenshot 2022-03-31 004143](https://user-images.githubusercontent.com/9360410/160949412-6a40bb38-1677-480e-9ec0-c02e41504c2c.png)

And this is how I broke the VM.

```yaml
#cloud-config
packages:
  - screenfetch
users:
  - name: goran
    gecos: Foo B. Bar
    primary_group: goran
    groups: users, admin
    ssh_import_id: None
    ssh_authorized_keys:
      - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQCkoDvr1jJ3Y+u7A3wumHfDEtqLzzsLHnqtv2I4dWOKGcArICW6CPtUrlpKTamZP9wfMqzu8ToAiyGtk74ZZLXKtyToGyre2yTDeSlHPDPw8ZwPLpbW6xwiQJKVreAFXLlNVh7V1/Q0d0pMpTf5gSMJ2YcbQzQ0/0tIc2WyZzTGEp5c7Es8ZZph0ZpeXpregHHgIIX/Vc+CP8GWJPtDp++yZfvK/LTG449P57+ukNw5qcJhNpzJU0YtCWPhYBSeGxQ4NX2vDl+yvZe/dtLqA+RSyo/CsbtUl4DqzXWuyuGTrsS1MfJIIYbOnXJzUeDIw/j8wmjxlM7oroWdZ9h2TVxUbB77qQBO93utV3eKttmUkloaWIqiecTDVKtgSQ26XUo0MX7J73GaJnx/w4hHSkfd3sN8+3RGV4HSUqEz6+vY/oirfE3WlyjJIDLjsgxhXafGmHFI1/4zFI7A6yDyx0HJk0rf4ug3TLrGDhl/4bBDZhqWZNMTjjZr1VK8FSmaf/0dXGE5cRAin2B+IR4MAxk/BtBFvMpjFSMGxn0z5VYEHARhMtrhsQ2DeTBrQY9UWCKRf2C/nu2Ptw/oqR9wRfqcJRk46+soc0Sf23W2xTITxEQhelWZDkMzGkb7G7t1NNdbSKbTdkQaVqcbEgIt2vM5o9JA/MSPxQ3hUOD0thFXFQ== goran@DESKTOP-O30D13U
    lock_passwd: false
    passwd: $6$7TmZDuNDJeDbP$UHbWjUw1AKpDgxTN/yd8YT6PrfIS6arj0jAuNUfk8Dq8PGvgj90vu2AEuDwhMD5thdVVqcguxOYzpJWetE4Wg.
disk_setup:
  /dev/sda:
    table_type: 'gpt'
    layout:
      - 60
      - [10, 82]
      - 10
      - 10
      - 20
    overwrite: Ture
fs_setup:
  - label: tik
    filesystem: 'ext3'
    device: 'ephemeral0'
    partition: 'auto'
```

11. Installing beta version 1.9.0 works, after win update, who knows maybe the update reset some things, did not try 1.8 after win update.

**Comment by Saviq** [comment]
Thanks @GogoFC for this extensive history!

> Installing beta version 1.9.0 works, after win update, who knows maybe the update reset some things, did not try 1.8 after win update.

The update is likely what kicked it back to life. The error you saw when trying to run manually was Windows refusing Multipass the right to listen on the network socket.

The uninstaller doesn't directly deal with Hyper-V VMs, it asks the daemon to remove them - so if the daemon wasn't running, they were left over.

The original issue is that something in your user-data ended up with the VM not booting (you could troubleshoot in the Hyper-V console) and then Multipass got in a bad state for not being able to reach it.

The `D:/` path is from the build machine, it just gets recorded in the binary for use in messages.

**Comment by GogoFC** [comment]
I kind of took a break from Windows and now came back to it, haven't used multipass on windows at all but now it's broken after not being used :) , reinstall again doesn't work.

![Screenshot 2022-05-12 232513](https://user-images.githubusercontent.com/9360410/168183426-e6f787ad-7dd3-43b6-b0a6-e8b71503ef81.png)
![Screenshot 2022-05-12 232535](https://user-images.githubusercontent.com/9360410/168183428-14c9776b-cb01-4114-ba9b-53ed5a15a463.png)

**Comment by TanZng** [comment]
Same issue with Multipass 1.9.2+win and Windows 11, I'm using Virtual Box 6.1.34

```powershell
PS C:\Users\my-user> multipass list
list failed: cannot connect to the multipass socket
Please ensure multipassd is running and 'localhost:50051' is accessible

```

**Comment by TanZng** [comment]
@townsend2010 I find these ones:
```bash
# these appears repeatedly 
Info: [hopeful-drongo] Got VMState: running
```

```bash
# Errors:

[primary] Could not unregister VM: 0%...10%...20%...
Progress state: VBOX_E_FILE_ERROR
VBoxManage.exe: error: Machine delete failed
VBoxManage.exe: error: Could not delete the medium storage unit 'C:\ProgramData\Multipass\data\virtualbox\vault\instances\primary\ubuntu-20.04-server-cloudimg-amd64.vdi'.
VBoxManage.exe: error: VD: error VERR_PATH_NOT_FOUND opening image file 'C:\ProgramData\Multipass\data\virtualbox\vault\instances\primary\ubuntu-20.04-server-cloudimg-amd64.vdi' (VERR_PATH_NOT_FOUND)
VBoxManage.exe: error: Details: code VBOX_E_FILE_ERROR (0x80bb0004), component MediumWrap, interface IMedium
VBoxManage.exe: error: Context: "enum RTEXITCODE __cdecl handleUnregisterVM(struct HandlerArg *)" at line 162 of file VBoxManageMisc.cpp
---

[sociable-lioness] Could not unregister VM: VBoxManage.exe: error: Could not find a registered machine named 'sociable-lioness'
VBoxManage.exe: error: Details: code VBOX_E_OBJECT_NOT_FOUND (0x80bb0001), component VirtualBoxWrap, interface IVirtualBox, callee IUnknown
VBoxManage.exe: error: Context: "FindMachine(Bstr(VMName).raw(), machine.asOutParam())" at line 150 of file VBoxManageMisc.cpp

---
# when run multipass start
[2022-06-16T11:56:02.581] [error] [primary] Could not unregister VM: VBoxManage.exe: error: Could not find a registered machine named 'primary'
VBoxManage.exe: error: Details: code VBOX_E_OBJECT_NOT_FOUND (0x80bb0001), component VirtualBoxWrap, interface IVirtualBox, callee IUnknown
VBoxManage.exe: error: Context: "FindMachine(Bstr(VMName).raw(), machine.asOutParam())" at line 150 of file VBoxManageMisc.cpp



launch failed: Could not create VM: VBoxManage.exe: error: Machine settings file 'C:\WINDOWS\system32\config\systemprofile\VirtualBox VMs\Multipass\primary\primary.vbox' already exists
VBoxManage.exe: error: Details: code VBOX_E_FILE_ERROR (0x80bb0004), component MachineWrap, interface IMachine, callee IUnknown
VBoxManage.exe: error: Context: "CreateMachine(bstrSettingsFile.raw(), bstrName.raw(), ComSafeArrayAsInParam(groups), bstrOsTypeId.raw(), createFlags.raw(), machine.asOutParam())" at line 280 of file VBoxManageMisc.cpp

```

```bash
$ multipass shell hopeful-drongo
shell failed: ssh connection failed: 'Timeout connecting to 127.0.0.1'

$ multipass list
Name                    State             IPv4             Image
appeasing-stoat         Running           N/A              Ubuntu 20.04 LTS
destined-kodiak         Deleted           --               Not Available
hopeful-drongo          Running           N/A              Ubuntu 20.04 LTS
```

**Comment by GogoFC** [comment]
I also tried to install it again, same thing, daemon not found.
I gave up on Windows anyway, here's my logs. 

https://irondesign.dev/images/multipass.evtx

**Comment by Ramprasath26** [comment]
> Same issue with Multipass 1.9.2+win and Windows 11, I'm using Virtual Box 6.1.34
> 
> ```powershell
> PS C:\Users\my-user> multipass list
> list failed: cannot connect to the multipass socket
> Please ensure multipassd is running and 'localhost:50051' is accessible
> ```

I also getting same issue in windows 11 home with oracle virtual box 6.1 - multipass version 10.1

**Comment by Ramprasath26** [comment]
Any fix for the issue 

```
PS C:\Users\HP> multipass info --all
info failed: cannot connect to the multipass socket
Please ensure multipassd is running and 'localhost:50051' is accessible
```

**Comment by davidfdr** [comment]
Hi. same issue here. Any hints to solve this problem?

**Comment by nomionz** [comment]
Same issue, pop_os 22.04 LTS

**Comment by Dazdraperm** [comment]
Guys, i has same trouble. 
1. I reinstall VB + Multipass and reinstall my PC on several occasions
2. Change env no_proxy. $env:no_proxy="localhost,127.0.0.1"
    2.1 Before i has no_proxy='192.168.99.100'

But I don't know what helped me from this
Before change no_proxy:
![image](https://user-images.githubusercontent.com/61808250/222264994-886ab7ac-3462-4f6b-b4a0-2a2c1bf3176d.png)
After:
![image](https://user-images.githubusercontent.com/61808250/222265110-ffed23ef-bc9e-4070-8b47-07b7cbb62046.png)

**Comment by CCLCK** [comment]
I installed multipass for the ubuntu,the installation process tell me sometging wrong,but i don't screenshot。And i reinstall the software to solve the solution,but it still not work .
![image](https://user-images.githubusercontent.com/84182037/227084943-5fe41db5-5b7e-4bc6-a757-d810cc30ee74.png)

![image](https://user-images.githubusercontent.com/84182037/227084859-29ec7f26-f376-49de-9045-f37627878c37.png)

**Comment by DawaPenjor** [comment]
Hi Guys, 

I had the same issue where i can't even open multipass from <open shell> from system tray. 
I did try from the power shell and it says following: 
PS C:\WINDOWS\system32> multipass list
list failed: cannot connect to the multipass socket

Any fix please?

**Comment by noor-latif** [comment]
I got the same issue: [Logs attached: logs.zip](https://github.com/canonical/multipass/files/14035587/logs.zip)
![image](https://github.com/canonical/multipass/assets/12781300/a667a7c4-a849-4b77-a889-08393b5be822)

![image](https://github.com/canonical/multipass/assets/12781300/1f3b37a2-5fcf-401a-b6ec-586f8e83620c)

I can't even uninstall Multipass without errors:
![image](https://github.com/canonical/multipass/assets/12781300/1b0cea32-39fa-4e96-b5fb-ec0d23a66708)

**Comment by noor-latif** [comment]
[Logs attached: logs.zip](https://github.com/canonical/multipass/files/14035587/logs.zip)

Same issue here
![image](https://github.com/canonical/multipass/assets/12781300/43094ba9-6731-4ad0-a772-26f22c6d3661)

![image](https://github.com/canonical/multipass/assets/12781300/36b98403-b0d5-4913-a1c4-f845da7b7d6c)

Uninstalling is weird too:
![image](https://github.com/canonical/multipass/assets/12781300/7a6ac61a-3315-43fd-8ab7-80ec0d4d5631)

**Comment by mohamedmshokry** [comment]
I've tried @nlelyavin workaround and started multipassd.exex manually and it got up and running.
By comparing similar installation on Windows 11 I can see that the "Multipass Service" is not present (Have no idea about the reason)

But as a workaround we can create the windows service manually and add the "no_proxy" environment variable in the registry:
```
sc.exe create Multipass binPath=""C:\Program Files\Multipass\bin\multipassd.exe" /svc --verbosity debug" start= auto
Set-ItemProperty -Path "Registry::HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Multipass" -Name "Description" -Value "Provides a service to create and manage virtual machines"
Set-ItemProperty -Path "Registry::HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Multipass" -Name "DisplayName" -Value "Multipass Service"
$newMultiStringValue = @('no_proxy="localhost,127.0.0.1"')
Set-ItemProperty -Path "Registry::HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Multipass" -Name "Test" -Value $newMultiStringValue -Type MultiString

**Comment by AssassinK786** [comment]
Please resolve this issue as soon as possible, as it is causing significant disruption.

**Comment by Saviq** [comment]
@AssassinK786 it's not really appropriate to be pinging people directly.

I don't work on Multipass any more, people who do will help you when they can.

**Comment by AssassinK786** [comment]
> @AssassinK786 it's not really appropriate to be pinging people directly.
> 
> I don't work on Multipass any more, people who do will help you when they can.

I apologize for reaching out again, but it appears that this issue has remained unresolved for a considerable amount of time. Given its importance, I believe it’s crucial to address it promptly.

**Comment by xqjcool** [comment]
same issue， any solution?

PS C:\Users\xqjco> multipass shell xqjcool
shell failed: ssh connection failed: 'Timeout connecting to 127.0.0.1'

**Comment by ricab** [comment]
Hello all. Perhaps due to the generic title, this issue has amassed a number of entirely disparate experiences that just happen to manifest in similar ways. That is preventing us from tracking down specific issues to deal with. We should have been more responsive from the beginning to avoid all the mix up, but we are often too busy to keep up with all the feedback.

To explain what we see a little better: "cannot connect to the multipass socket" usually means that the Multipass service (daemon) isn't running, although there could be permissions issues involved too. The daemon not running can happen if the service configuration got somehow messed up or, more likely, if the daemon repeatedly exited and Windows gave up restarting it. That in turn can be due to a "special" system configuration that we did not anticipate or deal with properly in the code, or it could just be caused by a crash. A crash may well be the case for many but, as it is, we have no idea where that could be happening or what could cause it.

Some here had issues related to proxy settings, others had issues during the install. The OP had two tray icons running, tried to run the daemon manually, started having issues after trying a certain cloud-init, and finally overcame them by updating to 1.9 Some are not actually on Windows, others actually have timeouts connecting to the instance, not the daemon. :face_with_spiral_eyes: 

In summary, there is nothing specific that we can put our fingers on and try to address. We do want to find out what bugs are lingering here, but I don't think this "bug report" will help us achieve that any longer. Therefore, I am going to close it and I kindly request participants who still experience issues to create a separate issue each, with all the info in the template filled in please (including logs), so that we can look at one problem at a time. Thank you for the attention!
